control_panel_api/migrations/0011_auto_20170927_1449.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.4 on 2017-09-27 14:49
+from __future__ import unicode_literals
+
+import control_panel_api.validators
+import datetime
+from django.conf import settings
+import django.core.validators
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+from django.utils.timezone import utc
+import django_extensions.db.fields
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('control_panel_api', '0010_auto_20170906_1501'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='app',
+            name='created_by',
+            field=models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL),
```

Comment:
Is it `CASCADE` the best strategy here?

It would feel a bit harsh/unpredictable to delete an app/s3bucket just because the user who created it was deleted?

Should we use the `SET_NULL` maybe? The annoyance would be that *potentially* `created_by` *may be* `None`/`NULL` so we'll have to guard against that when we'll start to use it.

Reply:
Good idea.

Reply:
I think this highlights the need to mark things (users) as deleted, rather than actually deleting them.

Reply:
@andyhd yes, I think long term (or also medium-term :D) that's probably the best approach 👍